### DISCUSSION TITLE
Do we have support for satellites other than GPS?

### DISCUSSION THREAD

**Discussion by CloneWith** [kylecorry31/Trail-Sense#2198] [New Feature Ideas]
asking just out of curiosity lol

My device model supports locating using GPS, Bei Dou, GLONASS and Galileo. Noticeably Trail Sense gets locations much slower than other APPs do, and it often fails to get correct location.

**Comment by kylecorry31** [comment]
Hi, yes Trail Sense has support other than GPS (any satellite system your phone supports). What it does not have that other apps have is network based location. 

What you will likely find is that apps that support network based location are accurate at home or in urban/suburban areas and are much faster to get a location fix than GPS. But when in the wilderness, network location is much more inaccurate and can lead to a few wildly incorrect locations before converging (plus it requires Internet access to update the network location cache). A decision was made early on in Trail Sense to only use GNSS since it is only designed for wilderness use and does not use the Internet.

There's also a setting in Trail Sense which requires a certain number of satellites for it to be considered a fix - you can disable that in Settings > GPS > Require satellite fix.

Can you elaborate more on how it fails to get the correct location for you?
- Where are you using it when this happens (ex. in the wilderness, inside a house, inside a building, on a city street, etc)?
- What happens in Trail Sense when this happens? (ex. shows a timeout, just shows wrong coordinates, etc)

**Reply by CloneWith** [reply]
yep I've tried some in different situations:

- worked just fine without network in open spaces
- slower in a building, especially altitude

looks like the issue comes from the device itself.

**Reply by kylecorry31** [reply]
Thanks for the clarification - that sounds expected. Trail Sense is not designed for use in buildings, and altitude especially will be hard to obtain in a building (it requires more quality satellite fixes than location to be accurate). Here's a similar issue regarding the weather monitor (altitude) while in buildings with more details: https://github.com/kylecorry31/Trail-Sense/issues/1833
